homeassistant/helpers/template.py

```diff
@@ -205,11 +167,53 @@ def ensure_valid(self):
         except jinja2.exceptions.TemplateSyntaxError as err:
             raise TemplateError(err)
 
+    def ensure_valid_without_conditionals(self):
+        """Return if template is valid."""
+        if self._compiled_code_without_conditionals is not None:
+            return
+
+        # Here we convert the conditions to print statements
+        # so we collecte entities that appear in code branches
+        # that would normally to be transversed.
+        template_without_conditionals = re.sub(
+            r"{%[ \t]+(?:if|elif)", "{% print", self.template
+        )
+        template_without_conditionals = re.sub(
+            r"{%[ \t]+(?:else|endif)", '{% print ""', template_without_conditionals
+        )
```

Comment:
This isn't as clean as I would have liked, but it seemed like it would be more resilient (since the syntax shouldn't be able to change) that overloading 
`def parse_if(self):` in 
https://github.com/pallets/jinja/blob/d9756e1acc19c1788d21a2d1076f8e6a0f55d6da/src/jinja2/parser.py#L213